hip-lang-config.cmake.in

```diff
@@ -68,24 +68,38 @@ include( "${CMAKE_CURRENT_LIST_DIR}/hip-lang-targets.cmake" )
 #get_filename_component cannot resolve the symlinks if called from /opt/rocm/lib/hip
 #and do three level up again
 get_filename_component(_DIR "${CMAKE_CURRENT_LIST_DIR}" REALPATH)
+# root directory of the hip library installation
 get_filename_component(_IMPORT_PREFIX "${_DIR}/../../../" REALPATH)
 
+# extract HIP_CLANG_ROOT from CMAKE_HIP_COMPILER
+find_program(CMAKE_HIP_COMPILER_WITH_PATH ${CMAKE_HIP_COMPILER})
```

Comment:
CMake doesnt set `CMAKE_HIP_COMPILER` to full path like other language compilers? Or is this because cmake is calling this early in the toolchain?

Reply:
`CMAKE_HIP_COMPILER` is not special, all the `CMAKE_<LANG>_COMPILER` may or may not store full path.

Reply:
I see then we probably need to do this in hip-config.cmake as well, as this assumes a absolute path for the CMAKE_CXX_COMPILER.